src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -102,6 +105,12 @@ public static DatasetTable tableModelToTable(TableModel tableModel) {
             .collect(Collectors.toList());
         datasetTable.primaryKey(primaryKeyColumns);
 
+        datasetTable.partitionStrategy(tableModel.getPartitionStrategy());
+        Optional<Column> partitionColumn = Optional.ofNullable(tableModel.getPartitionColumn())
+            .map(columnMap::get);
+        datasetTable.partitionColumn(partitionColumn.orElse(null));
```

Comment:
I think Optional should only be used for output parameters. Using it just so you can do its streaming function seems to me to be a way to obscure the logic. How about:
```suggestion
if (tableModel.getPartitionColumn() != null) {
    datasetTable.partitionColumn(columnMap.get(tableModel.getPartitionColumn());
}
```